Illegal function pointer to void* cast.

https://github.com/NVIDIA/cub/blob/main/cub/util_device.cuh#L369

cudaFuncGetAttribute's 2nd parameter is const void*. Implicit casts from pointers to functions to void* are disallowed. You need an implicit cast.

```cpp
void func(const void*) { }

int main() {
  typedef void (*EmptyKernelPtr)();
  EmptyKernelPtr empty_kernel;

  func(empty_kernel);
}
```
```
call.cxx:7:3: error: no matching function for call to 'func'
  func(empty_kernel);
  ^~~~
call.cxx:1:6: note: candidate function not viable: no known conversion from 'EmptyKernelPtr' (aka 'void (*)()') to 'const void *' for 1st argument; take the address of the argument with &
void func(const void*) { }
     ^
```

I don't know why this compiles currently.

--- a/cub/util_device.cuh
+++ b/cub/util_device.cuh
@@ -371,11 +371,9 @@ CUB_RUNTIME_FUNCTION inline cudaError_t PtxVersionUncached(int& ptx_version)
        #if CUB_INCLUDE_HOST_CODE
             cudaFuncAttributes empty_kernel_attrs;
 
-            do {
-                if (CubDebug(result = cudaFuncGetAttributes(&empty_kernel_attrs, empty_kernel)))
-                    break;
-            }
-            while(0);
+            result = cudaFuncGetAttributes(&empty_kernel_attrs,
+                                           reinterpret_cast<void*>(empty_kernel));
+            CubDebug(result);
 
             ptx_version = empty_kernel_attrs.ptxVersion * 10;
         #endif

